.travis.yml

```diff
@@ -4,7 +4,16 @@ install:
 - pip install --user mkdocs mkdocs-material
 script:
 - mvn test -B
-- if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]; then cd documentation && mkdocs build --clean --verbose --strict && cd ..; fi;
+- |
+  if [[ "${TRAVIS_PULL_REQUEST}" != "false" ]]
+  then
+    cd documentation && mkdocs build --clean --verbose --strict && cd ..
+  else
+    if [ "${TRAVIS_BRANCH}" == "master" ]
+    then
+      gh-pages-deploy
```

Comment:
Are you sure you don't have to make this file executable before?

Reply:
I adjusted the file mode for gh-pages-deploy to 755. I gets checked out from git with the executable flag already set.

But I see now something else: should be ```./gh-pages-deploy``` rather than simply ```gh-pages-deploy```.